adding pending tasks, stats and reroute cluster apis docs (#10311)
opensearch-project/documentation-website: _api-reference/cluster-api/cluster-reroute.md

- `master_node`: Shows which node is currently acting as cluster manager.
- `metadata`: Returns index settings, mappings, and aliases. If specific indexes are targeted, only their metadata is returned.
- `nodes`: Includes all nodes in the cluster and their metadata.
- `routing_table`: Returns the routing information for all shards and replicas.
- `version`: Displays the cluster state version number.

You can combine values in a comma-separated list, such as `metric=metadata,nodes,routing_table`.

## Request body fields

The `commands` array in the request body defines actions to apply to shard allocation. Supported actions include:

### Move

Moves a started shard (primary or replica) from one node to another. This can be used to balance load or drain a node before maintenance. The shard must be in the `STARTED` state. Both primary and replica shards can be moved using this command. 

`move` command requires the following parameters:

* `index`: The name of the index.
* `shard`: Shard number.
* `from_node`: Name of the node to move the shard from.
* `to_node`: Name of the node to move the shard to.

### Cancel

> Cancels allocation of a shard (including recovery). This command forces resynchronization by canceling existing allocations and letting the system reinitialize them. Replica shard allocations can be canceled by default, but canceling a primary shard requires `allow_primary=true` to prevent accidental data disruption

Reviewer: ```suggestion
The `cancel` command cancels allocation of a shard (including recovery). This command forces resynchronization by canceling existing allocations and letting the system reinitialize them. Replica shard allocations can be canceled by default, but canceling a primary shard requires `allow_primary=true` to prevent accidental data disruption
```

`cancel` command requires the following parameters:

* `index`: The name of the index.
* `shard`: The shard number.
* `node`: The name or node ID of the node to perform action on.

> * `allow_primary` (optional: If `true`, allows cancelling primary shard allocations. Default is `false`).

Reviewer: ```suggestion
* `allow_primary` (_Optional): If `true`, allows cancelling primary shard allocations. Default is `false`.
```

### Allocate_replica

> Assigns an unassigned replica to a specified node. This operation respects allocation deciders. Use this command to manually trigger allocation of replicas when automatic allocation fails.

Reviewer: ```suggestion
The `allocate_replica` command assigns an unassigned replica to a specified node. This operation respects allocation deciders. Use this command to manually trigger allocation of replicas when automatic allocation fails.
```

> `allocate_replica` command requires the following parameters:

Reviewer: ```suggestion
The `allocate_replica` command requires the following parameters:
```

* `index`: The name of the index.
* `shard`: The shard number.

> * `node`: The name or node ID of the node to perform action on.

Reviewer: ```suggestion
* `node`: The name or node ID of the node to perform the action on.
```

> ### Allocate_stale_primary

Reviewer: ```suggestion
### Allocate stale primary
```

> Force-allocates a primary shard to a node that holds a stale copy. This command should be used with extreme caution. It bypasses safety checks and may lead to **data loss**, especially if a more recent shard copy exists on another node that is temporarily offline. If that node rejoins the cluster later, its data will be deleted or replaced by the stale copy that was forcefully promoted.

Reviewer: ```suggestion
The `allocate_stale_primary` command force-allocates a primary shard to a node that holds a stale copy. 

This command should be used with extreme caution. It bypasses safety checks and may lead to **data loss**, especially if a more recent shard copy exists on another node that is temporarily offline. If that node rejoins the cluster later, its data will be deleted or replaced by the stale copy that was forcefully promoted.
{: .warning}
```

> Use this only when no up-to-date copies are available and you have no way to restore the original data.

Reviewer: ```suggestion
Use this command only when no up-to-date copies are available and you have no way to restore the original data.
{: .tip}
```

> `allocate_stale_primary` command requires the following parameters:

Reviewer: ```suggestion
The `allocate_stale_primary` command requires the following parameters:
```

* `index`: The name of the index.
* `shard`: The shard number.

> * `node`: The name or node ID of the node to perform action on.

Reviewer: ```suggestion
* `node`: The name or node ID of the node to perform the action on.
```

> * `accept_data_loss`: Must be `true`

Reviewer: ```suggestion
* `accept_data_loss`: Must be set to `true`.
```

> ### Allocate_empty_primary

Reviewer: ```suggestion
### Allocate empty primary
```

> Force-allocates a new empty primary shard to a node. This operation initializes a new primary shard without any existing data. Any previous data for the shard will be **permanently lost**. If a node with valid data for that shard later rejoins the cluster, its copy will be erased. This command is intended for disaster recovery when **no valid shard copies exist** and recovery from snapshot or backup is not an option.

Reviewer: ```suggestion
The `allocate_empty_primary` command force-allocates a new empty primary shard to a node. This operation initializes a new primary shard without any existing data. 

Any previous data for the shard will be **permanently lost**. If a node with valid data for that shard later rejoins the cluster, its copy will be erased. This command is intended for disaster recovery when **no valid shard copies exist** and recovery from snapshot or backup is not an option.
{: .warning}
```

> `allocate_empty_primary` command required the following parameters:

Reviewer: ```suggestion
The `allocate_empty_primary` command requires the following parameters:
```

* `index`: The name of the index.
* `shard`: The shard number.

> * `node` : The name or node ID of the node to perform action on.

Reviewer: ```suggestion
* `node` : The name or node ID of the node to perform the action on.
```

> * `accept_data_loss`: Must be `true`

Reviewer: ```suggestion
* `accept_data_loss`: Must be set to `true`.
```

## Example

> See following examples using `reroute` API.

Reviewer: ```suggestion
The following are examples of using the Cluster Reroute API.
```

### Moving a shard

Create a sample index:

```json
PUT /test-cluster-index
{
  "settings": {
    "number_of_shards": 1,
    "number_of_replicas": 1
  }
}
```
{% include copy-curl.html %}

Run the following reroute command to move shard `0` of the index `test-cluster-index` from node `node1` to node `node2`:

```json
POST /_cluster/reroute
{
  "commands": [
    {
      "move": {
        "index": "test-cluster-index",
        "shard": 0,
        "from_node": "node1",
        "to_node": "node2"
      }
    }
  ]
}
```
{% include copy-curl.html %}

### Simulating a reroute

To simulate a reroute without executing it, set `dry_run=true`:

```json
POST /_cluster/reroute?dry_run=true
{
  "commands": [
    {
      "move": {
        "index": "test-cluster-index",
        "shard": 0,
        "from_node": "node1",
        "to_node": "node2"
      }
    }
  ]
}
```
{% include copy-curl.html %}

### Retrying failed allocations

> If some shards failed to allocate due to previous issues, you can reattempt allocation:

Reviewer: ```suggestion
If some shards failed to allocate because of previous issues, you can reattempt allocation:
```

```json
POST /_cluster/reroute?retry_failed=true
```

{% include copy-curl.html %}

### Explaining reroute decisions

To understand why a reroute command is accepted or rejected, add `explain=true`:

```json
POST /_cluster/reroute?explain=true
{
  "commands": [
    {
      "move": {
        "index": "test-cluster-index",
        "shard": 0,
        "from_node": "node1",
        "to_node": "node3"
      }
    }
  ]
}
```
{% include copy-curl.html %}

This returns a `decisions` array explaining the outcome:

```json
"decisions": [
        {
          "decider": "max_retry",
          "decision": "YES",
          "explanation": "shard has no previous failures"
        },
        {
          "decider": "replica_after_primary_active",
          "decision": "YES",
          "explanation": "shard is primary and can be allocated"
        },
        ...
        {
          "decider": "remote_store_migration",
          "decision": "YES",
          "explanation": "[none migration_direction]: primary shard copy can be relocated to a non-remote node for strict compatibility mode"
        }
      ]
```

## Response body fields

> The response includes cluster state metadata and optionally a `decisions` array if `explain=true` was used.

Reviewer 2: ```suggestion
The response includes cluster state metadata and, optionally, a `decisions` array if `explain=true` was used.
```

| Field                        | Data type | Description                                                             |
| ---------------------------- | --------- | ----------------------------------------------------------------------- |
| `acknowledged`               | Boolean   | States whether the reroute request was acknowledged.                           |
| `state.cluster_uuid`         | String    | The unique identifier of the cluster.                                       |
| `state.version`              | Integer   | The version of the cluster state.                                           |
| `state.state_uuid`           | String    | The UUID for this specific state version.                                   |

> | `state.master_node`          | String    | Same as `cluster_manager_node`, this is maintained for backward compatibility.                                 |

Reviewer 2: ```suggestion
| `state.master_node`          | String    | As with `cluster_manager_node`, this is maintained for backward compatibility.                                 |
```

| `state.cluster_manager_node` | String    | The ID of the elected cluster manager node.  |
| `state.blocks`               | Object    | Any global or index-level cluster blocks.                               |

> | `state.nodes`                | Object    | The cluster nodes metadata, including its name and address.                      |

Reviewer 2: ```suggestion
| `state.nodes`                | Object    | The cluster node's metadata, including its name and address.                      |
```
